File: java/program51.java
import java.util.*;

class StringX
{
    public String str;

    StringX(String s)
    {
        this.str = s;
    }

    public int SearchFirstOcc(char ch)
    {
        char Arr[] = str.toCharArray();
        int i = 0;

        while(i < Arr.length)
        {
            if(Arr[i] == ch)
            {
                break;
            }
            i++;
        }
        if(i == Arr.length)
        {
            return -1;
        }
        else
        {
            return i;
        }
    }
}

class program51
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter string");
        String str = sobj.nextLine();

        System.out.println("Enter the character you want to search");
        char ch = sobj.next().charAt(0);     

        StringX obj = new StringX(str);
        int iRet = obj.SearchFirstOcc(ch);
        
        if(iRet != -1)
        {
            System.out.println("First occurence of the character is at index : "+iRet);
        }
        else
        {
            System.out.println("There is no such character");
        }
    }
}

File: cpp/program51.cpp
// program173.c

#include<iostream>
using namespace std;

#define ERR_NOTFOUND -1

class String
{
    public:

        int SearchFirstOcc(char *str, char cValue)
        {
            int iCnt = 0;

            while((*str != '\0') && (*str != cValue))
            {
                iCnt++;
                str++;
            }

            if(*str == '\0')
            {
                return ERR_NOTFOUND;
            }
            else
            {
                return iCnt;
            }
        }
};

int main()
{
    char Arr[20];
    char ch = '\0';
    int iRet = 0;

    cout<<"Enter string \n";
    cin>>Arr;

    cout<<"Enter the character you want to search"<<"\n";
    cin>>ch;

    String obj;

    iRet =obj.SearchFirstOcc(Arr,ch);
    if(iRet == ERR_NOTFOUND)
    {
        cout<<"There is no such character \n";
    }
    else
    {
        cout<<"First occurence is at index : "<<iRet<<"\n";
    }
    

    return 0;
}